### ISSUE TITLE
[Feature Request]: launch terminal in program output window

### ISSUE THREAD

**Issue by darc1** [puremourning/vimspector#195]
**Is your feature request related to a problem? Please describe.**
The terminal window makes the code window very small and since there is no way to configure scaling of the windows I need to resize them manually every time I run the debugger

**Describe the solution you'd like**
it will be great if it can be a tab in the output window.

**Describe alternatives you've considered**
option to hide/show windows.
making the program output window smaller can mitigate the issue.

**Additional context**
<img width="1667" alt="Screen Shot 2020-07-09 at 13 52 13" src="https://user-images.githubusercontent.com/10958651/87031344-acdf1d80-c1eb-11ea-8ee1-1c95767029bc.png">

**Comment by puremourning** [comment]
i think that's possible. I hadn't considered it.

Not a completely simple change though and might be subjective. certainly you might want to see the program output _at the same time_ as the interactive debug console.

**Comment by darc1** [comment]
I think that the major issue for me is that the code window is very small, I think it should support at least 80 columns (120 even better), I find it hard to debug that way.
if adding it to the output is difficult, I would have preferred to have the terminal as a split of the output window and having the output window height to match the stack trace window height, but again this is subjective. 
tried to tweak the params in debug_session.py but it didn't help.

**Comment by puremourning** [comment]
> I think that the major issue for me is that the code window is very small, I think it should support at least 80 columns (120 even better), I find it hard to debug that way.

It's only small because of the size of your screen on my screen, it's plenty big enough.

> tried to tweak the params in debug_session.py to reduce the height of the output window and the width of the watch/variables/stacktrace but it didn't help. any hint on how can I define the scales in the code :)?

https://github.com/puremourning/vimspector/blob/master/python3/vimspector/debug_session.py#L459

There's also this, which is where i'm putting all "how do i customise the UI" requests: https://github.com/puremourning/vimspector/issues/86

**Comment by darc1** [comment]
what is the recommended screen size, resolution, and font size?

**Comment by puremourning** [comment]
well the current UI code puts 10 columns for the left hand side and the rest is used for the code window. when a run in terminal request is issued, then it vertically splits the code window, so 10 + 2x your cold window desired size (e.g. 80*2 + 10).

As i said, UI customisation isn't possible yet, so this is the best i have for now.

**Comment by puremourning** [comment]
by the way, you can adjust the size of the terminal window by settings `termwinsize`

I haven't tried it, but something like this will restrict the width of the terminal window to 80 chars:

```
set termwinsize=0*80
```

You can also add this line `'term_cols': 80` in `code.py` linked above to restrict the width:

```
    options = {
      'vertical': 1,
      'term_cols': 80,
      'norestore': 1,
      'cwd': cwd,
      'env': env,
    }
```

**Comment by puremourning** [comment]
I added the above in pr #197

**Comment by puremourning** [comment]
I added the ability to customise the terminal window in #197 , example:

```viml
execute 'source' expand( '<sfile>:p:h' ) . '/minimal_vimrc'
set noequalalways

function! s:CustomiseUI()
  let wins = g:vimspector_session_windows

  " Close the Variables window
  if has( 'nvim' )
    " No win_execute in neovim
    call win_gotoid( wins.variables )
    quit
  else
    call win_execute( wins.variables, 'q' )
  endif

  " Put the stack trace at the top of the "left bar" (rotate)
  call win_gotoid( wins.stack_trace )
  wincmd r

  " Make the left column at least 70 chars
  70wincmd |

  " Make the output window 10 lines high and right at the top of the screen
  call win_gotoid( wins.output )
  10wincmd _
  wincmd K
endfunction

function s:SetUpTerminal()
  let terminal_win = g:vimspector_session_windows.terminal

  " Make the terminal window at most 80 columns wide, ensuring there is enough
  " sapce for our code window (80 columns) and the left bar (70 columns)

  " Padding is 2 for the 2 vertical split markers and 2 for the sign column in
  " the code window.
  let left_bar = 70
  let code = 80
  let padding = 4
  let cols = min([ &columns - left_bar - code - padding, 80 ] )

  " But let's have a minimum of say 10
  if cols >= 10
    call win_gotoid( terminal_win )
    execute cols . 'wincmd |'
  endif
endfunction

augroup TestUICustomistaion
  autocmd!
  autocmd User VimspectorUICreated call s:CustomiseUI()
  autocmd User VimspectorTerminalOpened call s:SetUpTerminal()
augroup END
```

**Comment by darc1** [comment]
cool, will check it out.
I think I have another issue, the terminal windows are always split equally (see screenshot in the original post).

**Comment by puremourning** [comment]
per that example, you will be able to resize it with an autocommand.

**Comment by darc1** [comment]
where are the definitions of `User VimspectorUICreated` and `User VimspectorTerminalOpend`?

**Comment by darc1** [comment]
I added this to my `.vimrc`
```vim                                                                                                                                                                               
execute 'source' g:plug_home. '/vimspector/support/custom_ui_vimrc'
```

this really messed up the debugger view:

<img width="1673" alt="Screen Shot 2020-07-13 at 13 54 15" src="https://user-images.githubusercontent.com/10958651/87297017-6e5da180-c510-11ea-961c-67a6a9ead0b9.png">

**Comment by puremourning** [comment]
> EDIT: the issue was that I didn't have set noequalalways in my vimrc

Yeah, i think there might be a bug for me to look into when `equalalways` is set. I try to disable it while generating the UI but it seems that Vim still applies it. I'll try and fix that with pr #198
